pkg/tests/tasks/security/certificate/external_ca_test.go

```diff
@@ -15,109 +15,193 @@
 
 	"github.com/maistra/maistra-test-tool/pkg/app"
-	"github.com/maistra/maistra-test-tool/pkg/util"
 	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
 	"github.com/maistra/maistra-test-tool/pkg/util/curl"
 	"github.com/maistra/maistra-test-tool/pkg/util/env"
 	"github.com/maistra/maistra-test-tool/pkg/util/oc"
 	"github.com/maistra/maistra-test-tool/pkg/util/pod"
 	"github.com/maistra/maistra-test-tool/pkg/util/retry"
-	"github.com/maistra/maistra-test-tool/pkg/util/shell"
 	"github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
 func TestExternalCertificate(t *testing.T) {
 	test.NewTest(t).Id("T17").Groups(test.Full, test.ARM, test.InterOp).Run(func(t test.TestHelper) {
 		const ns = "bookinfo"
 
-		var tmpDir = ""
-
 		t.Cleanup(func() {
-			if tmpDir != "" {
-				os.RemoveAll(tmpDir)
-			}
-
+			t.Logf("Recreate namespace %s", ns)
 			oc.RecreateNamespace(t, ns, meshNamespace)
 		})
 
 		t.LogStep("Uninstall existing SMCP")
 		oc.RecreateNamespace(t, meshNamespace)
 
-		t.LogStep("Create cacerts Secret")
+		t.LogStep("Create cacerts secret")
 		oc.CreateGenericSecretFromFiles(t, meshNamespace, "cacerts",
 			"ca-cert.pem="+sampleCACert,
 			"ca-key.pem="+sampleCAKey,
 			"root-cert.pem="+sampleCARoot,
 			"cert-chain.pem="+sampleCAChain)
+		rootCert := readPemCertificatesFromFile(t, sampleCARoot)[0]
+		chainCerts := readPemCertificatesFromFile(t, sampleCAChain)
 
-		t.LogStep("Apply SMCP to configure certificate authority to use cacerts Secret")
+		t.LogStep("Apply SMCP to configure certificate authority to use cacerts secret")
 		oc.ApplyString(t, meshNamespace, createSMCPWithCustomCert(smcpName, ns))
 		oc.WaitSMCPReady(t, meshNamespace, smcpName)
 
 		t.LogStep("Install bookinfo")
 		app.InstallAndWaitReady(t, app.BookinfoWithMTLS(ns))
 
-		t.LogStep("Checking response from productpage.")
+		t.LogStep("Wait for response from productpage app")
 		retry.UntilSuccess(t, func(t test.TestHelper) {
 			curl.Request(t,
 				app.BookinfoProductPageURL(t, meshNamespace), nil,
 				assert.ResponseStatus(200))
 		})
 
-		if env.GetArch() == "p" || env.GetArch() == "z" {
-			t.Log("NOTE: Not checking certificates, because test is running in P or Z environment")
-			return
-		}
+		t.LogStep("Retrieve certificates from bookinfo details service")
+		opensslOutput := oc.Exec(t, pod.MatchingSelector("app=productpage", ns), "istio-proxy",
+			`openssl s_client -showcerts -connect details:9080 || true`) // the "|| true" is needed until oc.Exec can ignore return codes
+		returnedCerts := readPemCertificatesFromText(t, opensslOutput)
+		certToVerify := returnedCerts[0]
+		otherCerts := returnedCerts[1:]
+
+		t.NewSubTest("Verify correct certificates are received").Run(func(t test.TestHelper) {
+			verifyContainsCerts(t, otherCerts, chainCerts,
+				"The cert-chain certificates are present in the certificates sent by the tested service",
+				"Error: The cert-chain certificates were not found in the certificates sent by the tested service")
```

Comment:
No need to prefix error messages with "Error", as this is already handled by `t.Error/Fatal`

Reply:
I have re-worded all the error messages in the test itself, but are the error messages in my helper functions ok? When the helper itself breaks, it usually generates a message like "Error while doing X: [error details]".